configure.ac

```diff
@@ -194,38 +194,32 @@ AS_IF([test x$enable_staged = xyes], [
   AC_DEFINE(BUILDOPT_STAGE_DEPLOYMENTS, 1, [Define if we are staging deployments])
 ])
 
-AC_ARG_ENABLE(rust,
-              AS_HELP_STRING([--enable-rust],
-                             [Compile Rust features (e.g. compose tree --yaml)]),,
-              [enable_rust=no; rust_debug_release=no])
+AC_PATH_PROG([cargo], [cargo])
+AS_IF([test -z "$cargo"], [AC_MSG_ERROR([cargo is required for --enable-rust])])
+AC_PATH_PROG([rustc], [rustc])
+AS_IF([test -z "$rustc"], [AC_MSG_ERROR([rustc is required for --enable-rust])])
+AC_DEFINE(HAVE_RUST, 1, [Define if we are building with Rust])
 
-AS_IF([test x$enable_rust = xyes], [
-    AC_PATH_PROG([cargo], [cargo])
-    AS_IF([test -z "$cargo"], [AC_MSG_ERROR([cargo is required for --enable-rust])])
-    AC_PATH_PROG([rustc], [rustc])
-    AS_IF([test -z "$rustc"], [AC_MSG_ERROR([rustc is required for --enable-rust])])
-    AC_DEFINE(HAVE_RUST, 1, [Define if we are building with Rust])
+dnl These bits based on gnome:librsvg/configure.ac
+dnl By default, we build in public release mode.
+AC_ARG_ENABLE(rust-debug,
+  AC_HELP_STRING([--enable-rust-debug],
```

Comment:
Could we do this conditionally on `CFLAGS` instead of an `--enable` option? I think we always want those in sync, right? (As in, if we compile debuginfo for one, we'd want it for the other as well?). Similar to what we do for libdnf below.

Reply:
This is a tricky topic because in Rust release vs debug are *very* different today as Rust heavily relies on inlining, to a much greater degree than (our) C.

Anyways, done :arrow_down:

Reply:
I also snuck in a reference to https://github.com/rust-lang/rust/issues/47714 that I found recently while googling for the CI failure on https://src.fedoraproject.org/rpms/rpm-ostree/pull-request/22